fix(transformer/class-properties): preserve computed keys if they may have side effects when `remove_class_fields_without_initializer` is enabled

Dunqing opened the pull request:
When `remove_class_fields_without_initializer` is enabled, fields without initializers would be removed. However, if the key is computed, and it may have side effects, then we should extract it and place it before class, so that it can keep the same behavior as before.

**Example:** 

Input:
```ts
import { Collection } from './collection.ts';

export class Obj {
  public readonly [Collection.identifier] = true;
  public readonly [Collection.identifier2];
  public static readonly [Collection.identifier3] = true;
  public static readonly  [Collection.identifier4];
}
```

Options:
```json
{
  "assumptions": {
    "setPublicClassFields": true
  },
  "plugins": [
    "transform-class-properties",
    [
      "transform-typescript",
      {
        "removeClassFieldsWithoutInitializer": true
      }
    ]
  ]
}
```

Output:
```js
import { Collection } from "./collection.ts";
let _Collection$identifie, _Collection$identifie2;
_Collection$identifie = Collection.identifier;
Collection.identifier2;
_Collection$identifie2 = Collection.identifier3;
Collection.identifier4;
export class Obj {
  constructor() {
    this[_Collection$identifie] = true;
  }
}
Obj[_Collection$identifie2] = true;
```

Dunqing commented:
* **#12772** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12772?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12772?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12772).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Aug 3, 7:15 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Aug 3, 7:15 AM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc).

Dunqing committed:
fix(transformer/class-properties): preserve computed keys if they may have side effects when `remove_class_fields_without_initializer` is enabled (#12772)

When `remove_class_fields_without_initializer` is enabled, fields without initializers would be removed. However, if the key is computed, and it may have side effects, then we should extract it and place it before class, so that it can keep the same behavior as before.

**Example:**

Input:
```ts
import { Collection } from './collection.ts';

export class Obj {
  public readonly [Collection.identifier] = true;
  public readonly [Collection.identifier2];
  public static readonly [Collection.identifier3] = true;
  public static readonly  [Collection.identifier4];
}
```

Options:
```json
{
  "assumptions": {
    "setPublicClassFields": true
  },
  "plugins": [
    "transform-class-properties",
    [
      "transform-typescript",
      {
        "removeClassFieldsWithoutInitializer": true
      }
    ]
  ]
}
```

Output:
```js
import { Collection } from "./collection.ts";
let _Collection$identifie, _Collection$identifie2;
_Collection$identifie = Collection.identifier;
Collection.identifier2;
_Collection$identifie2 = Collection.identifier3;
Collection.identifier4;
export class Obj {
  constructor() {
    this[_Collection$identifie] = true;
  }
}
Obj[_Collection$identifie2] = true;
```
